### PR TITLE
(#1133) MaxOf doesn't work with negative numbers

### PR DESCRIPTION
**Pull request by fabriciofx**
For #1135 
- Fix `MaxOf` to work properly with negative numbers
- `MinOf` doesn't need fix.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1153](https://github.com/yegor256/cactoos/issues/1153) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1153](https://github.com/yegor256/cactoos/issues/1153) is assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), here is [why](https://www.0crat.com/footprint/C63314D6Z/b3fc3b1f-9043-44dd-8ed6-3c59efcb5366); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by fabriciofx** [comment]
@0crat wait for `REV`.

**Comment by 0crat** [comment]
> @0crat wait for `REV`. ([here](https://github.com/yegor256/cactoos/issues/1153#issuecomment-507375034))

@fabriciofx @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) you can't register impediment for this job

**Comment by fabriciofx** [comment]
@llorllale fixed. Can you take a look, please?

**Comment by fabriciofx** [comment]
@llorllale ping

File: src/test/java/org/cactoos/scalar/MaxOfTest.java [hunk 1]
```diff
@@ -39,114 +38,193 @@
...
-            Matchers.equalTo(4.0d)
-        );
-        MatcherAssert.assertThat(
-            new MaxOf(
-                new ListOf<>(1, 2, 3, 4).toArray(new Integer[4])
-            ).floatValue(),
-            Matchers.equalTo(4.0f)
-        );
+        new Assertion<>(
+            "must maximum integer of positive integers",
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/MaxOfTest.java]: @fabriciofx the assertion messages need to be fixed. Eg. this one should read "must select maximum integer of positive integers"

**Comment by llorllale** [comment]
@fabriciofx left you a comment

Also, I'm not sure: are there any tests with a mix of positive and negative numbers?

File: src/test/java/org/cactoos/scalar/MaxOfTest.java (hunk 1 above)
Comment (fabriciofx) [reply src/test/java/org/cactoos/scalar/MaxOfTest.java]: @llorllale done.

**Comment by fabriciofx** [comment]
@llorllale added tests for positive and negative numbers. Please, take a look.

**Comment by fabriciofx** [comment]
@llorllale ping

**Comment by fabriciofx** [comment]
@llorllale I'm sorry, but the build fail is annoyed. We must fix it ASAP.

**Comment by llorllale** [comment]
@fabriciofx 

> [INFO] PMD: src/test/java/org/cactoos/scalar/MaxOfTest.java[37-326]: This class has too many methods, consider refactoring it. (TooManyMethods)

Suppress this check in your PR

**Comment by fabriciofx** [comment]
@llorllale done.

**Comment by fabriciofx** [comment]
@llorllale ping

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#1153](https://github.com/yegor256/cactoos/issues/1153) is not in the agenda of @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (16 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#1153](https://github.com/yegor256/cactoos/issues/1153) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by fabriciofx** [comment]
@sereshqua I'm sorry, but why the quality was bad?

**Comment by sereshqua** [comment]
@fabriciofx Fabric, the thing is, the CR was assigned to @vzurauskas, but he did not complete it at all :) so not job was done

**Comment by fabriciofx** [comment]
@sereshqua oh-oh! You're completely right! Sorry! :)

**Pull request merged**